### PR TITLE
Channel index fix

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
ES doesn't update an index if it already exists. So we're purging the channel's index before updating it.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-49777

EE PR - https://github.com/mattermost/enterprise/pull/1420

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/migrations.go [hunk 1]
```diff
@@ -559,6 +559,18 @@ func (s *Server) doPostPriorityConfigDefaultTrueMigration() {
...
 }
 
+func (s *Server) doElasticsearchFixChannelIndex() {
+	// If the migration is already marked as completed, don't do it again.
+	if _, err := s.Store().System().GetByName(model.MigrationKeyElasticsearchFixChannelIndex); err == nil {
+		return
+	}
+
+	if _, appErr := s.Jobs.CreateJob(model.JobTypeElasticsearchFixChannelIndex, nil); appErr != nil {
+		mlog.Error("failed to start job for fixing Elasticsearch channels index", mlog.Err(appErr))
```
Comment (agnivade) [inline server/channels/app/migrations.go:569]: Let's use `mlog.Fatal` here to remain consistent with other migrations.

Comment (harshilsharma63) [reply server/channels/app/migrations.go]: Done.

**Review by agnivade — APPROVED** [review]

**Review by koox00 — APPROVED** [review]

**Comment by amyblais** [comment]
@harshilsharma63 Will this need QA review?

**Comment by harshilsharma63** [comment]
I'm merging this as this can get stale easily. Also, no QA required IMO.

**Pull request merged**